src/cdk-experimental/menu/focus-mouse-manager.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {QueryList, ElementRef} from '@angular/core';
+import {fromEvent, Observable} from 'rxjs';
+import {mapTo, mergeAll, takeUntil, startWith, mergeMap} from 'rxjs/operators';
+
+/** Item to track for mouse focus events. */
+export interface FocusableElement {
+  /** A reference to the element to be tracked. */
+  _elementRef: ElementRef<HTMLElement>;
+}
+
+/**
+ * FocusMouseManager keeps track of the element under mouse focus. It emits the element on an
+ * observable when an element is moused into.
```

Comment:
After our conversation, I realized this could just be a function rather than a class since it doesn't have any state and just has a single method
```ts
/**
 * Gets a stream of pointer (mouse) entries into the given items.
 * This should typically run outside the Angular zone. 
 */
function getItemPointerEntries<T extends FocusableElement>(items: QueryList<T>): Observable<T> {
  return this._elements.changes.pipe(
      startWith(this._elements),
      mergeMap((list: QueryList<T>) =>
          list.map(element =>
              fromEvent(element._elementRef.nativeElement, 'mouseenter').pipe(
                  mapTo(element),
                  takeUntil(this._elements.changes)
              )
          )
      ),
      mergeAll()
  );
}
```